### PR TITLE
Implement explicit ping handling

### PR DESCRIPTION
**Pull request by Foelsgaard**
Adds two features for explicit ping handling. A `ping-response` feature for adding a `PingResponse` message and a default feature `automatic-pinging` that adds the ping thread to the mqtt transport.

### CODE REVIEW

File: src/mqtt_transport.rs [hunk 1]
```diff
@@ -218,35 +205,19 @@ impl MqttTransport {
...
             // rx_loop_handle: None,
         };
 
-        mqtt_transport.ping_join_handle = Some(Arc::new(mqtt_transport.ping_on_secs_interval(8)));
-
         Ok(mqtt_transport)
     }
-
-    ///
-    fn ping_on_secs_interval(&self, ping_interval: u8) -> JoinHandle<()> {
```
Comment (laumann) [inline src/mqtt_transport.rs:239]: Who's pinging now then?

Comment (Foelsgaard) [reply src/mqtt_transport.rs:239]: The user, manually.

Comment (Foelsgaard) [reply src/mqtt_transport.rs:239]: There is a `ping` method on `IoTHubClient`, so they'd just call that one.

**Comment by Foelsgaard** [comment]
Maybe this whole thing should be gated behind a feature?

File: src/mqtt_transport.rs (hunk 1 above)
Comment (laumann) [reply src/mqtt_transport.rs:239]: OK, so with "automatic-ping" on, we ask the library to set up a pinging task for us - the difference now is that we will receive all the ping reponses in our client right?

File: src/mqtt_transport.rs [hunk 2]
```diff
@@ -218,6 +224,7 @@ impl MqttTransport {
 
         let (read_socket, write_socket) = tokio::io::split(socket);
 
+        #[allow(unused_mut)]
```
Comment (laumann) [inline src/mqtt_transport.rs:227]: alternatively I think you can do:

```rust
let mqtt_transport = Self {
    ...
};
#[cfg(feature = "automatic-ping")]
let mut mqtt_transport = mqtt_transport;

#[cfg(feature = "automatic-ping")]
{
    ...
}
```
I think that could also work.

**Review by laumann — APPROVED** [review]
I think this is good, will you open a PR against upstream?

**Pull request merged**